### DISCUSSION TITLE
How does Turn order Seating plugin work.

### DISCUSSION THREAD

**Discussion by ikeaton51** [jhoffmann2/OathUltimateCollection#27] [Q&A] [answered]
Having tried experimenting and looking through the code I can't figure out how the Turn Order Seating plugin, in the Oath Ultimate Collection mod, is supposed to work. Can you run me through the basics?

**Comment by jhoffmann2** [comment, marked as answer]
Answered in a discord call. Here's the gist: of our discussion

- Turn order seating doesn't manage turns themselves - that's handled by the Turns plugin and the RandomTurnOrder plugin. 
- The plugin's job is to order the player seats/boards so that they're in the same order as the TTS turns
- technically speaking, the OathOfficial mod is what actually does most of the setup. TurnOrderSeating just modifies some data that the official mod uses to know how to set up player boards
    - the data being modified exists in a "magic" variable that all plugins have called "shared". The shared object can store any data which  is reflected in all other scripts running on the same object (or in this case, running as global)
